nsdb-core/src/main/scala/io/radicalbit/nsdb/post_proc/package.scala

```diff
@@ -368,23 +367,25 @@ package object post_proc {
 
     val uniqueValues = rawResults.foldLeft(Set.empty[NSDbType])((acc, b2) => acc ++ b2.uniqueValues)
 
-    val allAggregationReduce =
-      if (finalStep && uniqueValues.nonEmpty)
-        aggregationsReduced + (`count(distinct *)` -> NSDbLongType(uniqueValues.size))
-      else aggregationsReduced
+    //only one count distinct is allowed. This has been checked previously in the flow
+    val allAggregationReduced = aggregations.find(_.isInstanceOf[CountDistinctAggregation]) match {
+      case Some(aggregation) if finalStep =>
```

Comment:
`only one count distinct is allowed. This has been checked previously in the flow`.
Is the above statement ensuring we never have `case Some(aggregation) => // finalStep = false`?

Reply:
That comment is related to a previous check performed by the semantic parser. There was a typo at the next line. So `case None ` has been updated to `case _`.